### PR TITLE
Add TextLinkButton component

### PR DESCRIPTION
**Pull request by markdalgleish**
See changeset for details.

### CODE REVIEW

File: generate-component-docs/__snapshots__/contract.test.ts.snap [hunk 1]
```diff
@@ -5991,6 +5991,22 @@ Object {
 }
 `;
 
+exports[`TextLinkButton 1`] = `
+Object {
+  exportType: component,
+  props: {
+    children: ReactNode
+    data?: Record<string, string | number>
+    hitArea?:
```
Comment (michaeltaranto) [inline generate-component-docs/__snapshots__/contract.test.ts.snap:6000]: This prop is redundant on this "version" of a link right?

File: generate-component-docs/__snapshots__/contract.test.ts.snap [hunk 2]
```diff
@@ -5991,6 +5991,22 @@ Object {
...
+  exportType: component,
+  props: {
+    children: ReactNode
+    data?: Record<string, string | number>
+    hitArea?: 
+        | "large"
+        | "standard"
+    id?: string
+    onClick?: () => void
+    showVisited?: boolean
```
Comment (michaeltaranto) [inline generate-component-docs/__snapshots__/contract.test.ts.snap:6005]: Does this still make sense?

Comment (markdalgleish) [reply generate-component-docs/__snapshots__/contract.test.ts.snap:6005]: Good catch 👍

File: generate-component-docs/__snapshots__/contract.test.ts.snap (hunk 1 above)
Comment (michaeltaranto) [reply generate-component-docs/__snapshots__/contract.test.ts.snap:6000]: Ignore me, wrong mental model.

File: generate-component-docs/__snapshots__/contract.test.ts.snap (hunk 2 above)
Comment (michaeltaranto) [reply generate-component-docs/__snapshots__/contract.test.ts.snap:6005]: Again ignore this

Comment (markdalgleish) [reply generate-component-docs/__snapshots__/contract.test.ts.snap:6005]: No, I think you're right on this. A button can't be marked as visited.

**Pull request merged**